server/src/test/java/io/crate/integrationtests/SQLTypeMappingTest.java

```diff
@@ -463,6 +463,40 @@ public void test_dynamic_null_array_overridden_to_integer_becomes_null() throws
         );
     }
 
+    public void test_dynamic_null_array_within_objects() {
+        execute("create table tbl (a int primary key, o object (dynamic)) with (column_policy = 'dynamic')");
+        execute("insert into tbl (a, o, n1, n2) values (1, {x=[],y=[null]}, [], [null,null])");
+        execute("refresh table tbl");
+        execute("select _doc from tbl");
+        assertThat(response).hasRows("""
+            {a=1, n1=[], n2=[null, null], o={x=[], y=[null]}}""");
+        execute("insert into tbl (a, o, n1, n2) values (2, {x=[1],y=[1]}, [1], [2])");
+        execute("refresh table tbl");
+        execute("select _doc from tbl order by a");
+        assertThat(response).hasRows("""
+                {a=1, n1=[], n2=[null, null], o={x=[], y=[null]}}""",
+            """
+                {a=2, n1=[1], n2=[2], o={x=[1], y=[1]}}""");
+    }
+
+    public void test_dynamic_nested_array_of_null_upcast() {
+        execute("create table t (a int) with (column_policy= 'dynamic')");
+        Asserts.assertSQLError(() -> execute("insert into t(a,b) values (1, [[]])"))
+                .hasMessageContaining("Dynamic nested arrays are not supported");
+    }
```

Comment:
There seems to be a way to work around this to created dynamic nested arrays:
```
cr> create table t (a int) with (column_policy= 'dynamic');
CREATE OK, 1 row affected (1.667 sec)
cr> insert into t(a,b) values (1, []);
INSERT OK, 1 row affected (0.161 sec)
cr> insert into t(a,b) values (1, [[]]);
INSERT OK, 1 row affected (0.162 sec)
```

Reply:
Sneaky.  Good catch, I will see if I can detect this case and throw.